sdk/tables/azure-data-tables/azure/data/tables/_table_client.py

```diff
@@ -441,25 +449,29 @@ def upsert_entity(  # pylint:disable=R1710
         entity = _add_entity_properties(entity)
 
         try:
+            metadata = None
             if mode is UpdateMode.MERGE:
-                self._client.table.merge_entity(
+                metadata, _ = self._client.table.merge_entity(
                     table=self.table_name,
                     partition_key=partition_key,
                     row_key=row_key,
                     table_entity_properties=entity,
+                    cls=kwargs.pop('cls', _return_headers_and_deserialized),
                     **kwargs
                 )
             elif mode is UpdateMode.REPLACE:
-                self._client.table.update_entity(
+                metadata, _ = self._client.table.update_entity(
                     table=self.table_name,
                     partition_key=partition_key,
                     row_key=row_key,
                     table_entity_properties=entity,
+                    cls=kwargs.pop('cls', _return_headers_and_deserialized),
                     **kwargs)
             else:
                 raise ValueError('Mode type is not supported')
```

Comment:
nit: for this error, I think you should have something like 

"Update mode {} is not supported. For a full list of supported modes, see the UpdateMode enum".format(mode)